cvat-ui/src/containers/annotation-page/standard-workspace/objects-side-bar/objects-list.tsx

```diff
@@ -184,27 +203,50 @@ class ObjectsListContainer extends React.PureComponent<Props, State> {
         this.state = {
             statesOrdering: StatesOrdering.ID_ASCENT,
             objectStates: [],
+            filteredStates: [],
             sortedStatesID: [],
         };
     }
 
-    static getDerivedStateFromProps(props: Props, state: State): State | null {
-        if (props.objectStates === state.objectStates) {
-            return null;
+    public componentDidMount(): void {
+        this.updateObjects();
+    }
+
+    public componentDidUpdate(prevProps: Props): void {
+        const {
+            statesSources, objectStates,
+        } = this.props;
+        const { objectStates: stateObjects } = this.state;
+        if (objectStates !== stateObjects ||
+            statesSources !== prevProps.statesSources
+        ) {
+            this.updateObjects();
```

Comment:
Probably ``getDerivedStateFromProps`` is exactly what you need here.

Reply:
I think its better to use DidUpdate, cuz we also need to know if statesSources changed